src/controllers/authentication.ts

```diff
@@ -169,6 +174,89 @@ export const checkPassword = async (req: Request, res: Response) => {
   }
 };
 
+export const forgotPassword = async (req: Request, res: Response) => {
+  try {
+    const user = await userModel.User.findOne({ email: req.params.email });
+
+    if (!user) {
+      return res.status(200).json({
+        message: `Email has been sent to ${req.params.email}`,
+      });
+    } else {
+      const token = jwt.sign({ id: user._id }, jwtSecret, {
+        expiresIn: maxAge,
+      });
+
+      const port = "5173";
+      const link = `http://127.0.0.1:${port}/reset-password/${user._id}/${token}`;
+
+      const transporter = nodemailer.createTransport({
+        service: "Gmail",
+        host: "smtp.gmail.com",
+        port: 3000,
+        secure: true,
+        auth: {
+          user: emailSender.email,
+          pass: emailSender.password,
+        },
+      });
+
+      const mailOptions = {
+        from: emailSender.email,
+        to: user.email,
+        subject: "Reset Password from Money Tracking",
+        text: `Hi ${user.name}, you have forgotten your password. Don't worry, just click on this link: ${link}. If you didn't request this, omit this email.`,
```

Comment:
This is still a plain text email, We need some html template to render a button where the link can be click and we have a styled email